streaming/storage/storage.go

```diff
@@ -45,15 +44,15 @@ func (bs *BadgerStorage) WithPrefix(prefix []byte) Storage {
 	return &copyStorage
 }
 
-func (bs *BadgerStorage) Subscribe(ctx context.Context, onNotification func() error) error {
-	cb := func(kv KeyValueList) error {
-		return onNotification()
-	}
-
-	err := bs.db.Subscribe(ctx, cb, bs.prefix)
-	if err != nil {
-		return errors.Wrap(err, "couldn't subscribe to db")
-	}
-
-	return nil
+func (bs *BadgerStorage) Subscribe(ctx context.Context) *Subscription {
+	return NewSubscription(ctx, func(ctx context.Context, changes chan<- struct{}) error {
+		return bs.db.Subscribe(ctx, func(kv *badger.KVList) error {
```

Comment:
Use KeyValueList defined above instead of *badger.KVList

Reply:
removed the type instead